### DISCUSSION TITLE
nextjs website deployment build failed

### DISCUSSION THREAD

**Discussion by pr0ta9** [vercel/community#6567] [Help]
### Summary

I got this error for my project:
>TypeError: Cannot read properties of undefined (reading 'User')
    at 3435 (/vercel/path0/.next/server/app/api/webhook/route.js:13:108698)
    at t (/vercel/path0/.next/server/edge-runtime-webpack.js:1:128)
    at /vercel/path0/.next/server/app/api/webhook/route.js:20:18362
    at r (/vercel/path0/.next/server/edge-runtime-webpack.js:1:1273)
    at /vercel/path0/.next/server/app/api/webhook/route.js:1:51
    at Script.runInContext (node:vm:148:12)
    at runInContext (node:vm:300:6)
    at evaluateInContext (/vercel/path0/node_modules/next/dist/server/web/sandbox/context.js:378:38)
    at getRuntimeContext (/vercel/path0/node_modules/next/dist/server/web/sandbox/sandbox.js:70:9)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> Build error occurred
Error: Failed to collect page data for /api/webhook
    at /vercel/path0/node_modules/next/dist/build/utils.js:1258:15
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  type: 'Error'
}
Error: Command "npm run build" exited with 1
I'm certain that the error is caused by this line: import User from '@models/user';
however, I tested locally and this line is pointing at the right location without any error, it is also in my other API file and it causes no problem. 

Deployment is fine when I remove import User from '@models/user'; but I need it for my MongoDB database access, any help is appreciated, thanks in advance!
This is the app directory:
...
-model
-src
--app
---api
----webhook

and this is my code for jsconfig.json:
{
  "compilerOptions": {
    "paths": {
      "@*": ["./*"]
    }
  }
}

### Steps to Reproduce

code for webhook route.js:
import Stripe from 'stripe';
import { connectToDB } from '@utils/database';
import User from '@models/user';

const stripe = new Stripe(process.env.STRIPE_SECRET_KEY, {
    apiVersion: '2023-10-16',
});

const endpointSecret = process.env.WEBHOOK_SECRET;

export const runtime = 'edge';

export const POST = async (req, res) => {
    try {
        const sig = req.headers.get('stripe-signature')
        const body = await req.text();
        let event;
        try {
            event = stripe.webhooks.constructEvent(body, sig, endpointSecret);
        } catch (e) {
            return new Response(`Webhook Error: ${e}`, {status:400});
        }
        if (event.type === 'checkout.session.completed') {
            await connectToDB();
            const session = await stripe.checkout.sessions.retrieve(
                event.data.object.id,
                {
                    expand: ["line_items"],
                }
            );
            const customer_id = session.customer;
            const lineItems = session.line_items;
            const plan_status = lineItems.data[0].description
            if (!lineItems) {
                return new Response(`Internal Server Error`, {status:500});
            }
            await User.updateOne({ email: event.data.object.customer_details.email }, { $set: {stripe_id: customer_id, user_plan: plan_status, has_plan: true} });
            return new Response(`SUCCESS`, {status:200});
        } else {
            return new Response(`Webhook Error.`, {status:400});
        }
    } catch (error) {
        return new Response(`Webhook Error: ${error}`, {status:400});
    }
};

**Comment by versecafe** [comment]
@Entrepreneurrichpj Could you share your git repo, without it it's hard to know but based off the error It seems like you may be trying to use User which is likely to not be importing correctly during build hence the undefined properties. It may be working in local development by a fluke sometimes things with the TS LSP can get into weird states with caching even when things are broken and will "build" but won't work on other systems, I'd recommend clearing your cache folders `.next` `.turbo` `dist` etc, restarting the LSP, and perhaps reinstalling `node_modules` just to make sure they aren't the source of any strange behaviour, after that see if it still builds locally, hopefully it should not and should give you some error messages in IDE to guide you such as incorrect import paths or an issue in `tsconfig.json` compiler options, paths section.

**Reply by pr0ta9** [reply]
I think you're right, there's some weird cache issue with User so it doesn't recognized my import, I was able to fix it by renaming it. Thank you for the help!
